src/resolvers/reactionResolver.ts

```diff
@@ -0,0 +1,260 @@
+import {
+  Arg,
+  Args,
+  Ctx,
+  Field,
+  Int,
+  Mutation,
+  ObjectType,
+  Query,
+  Resolver,
+} from 'type-graphql';
+import { Reaction, REACTION_TYPE } from '../entities/reaction';
+import { Context } from '../context';
+import { Project, ProjectUpdate } from '../entities/project';
+import { MyContext } from '../types/MyContext';
+import { errorMessages } from '../utils/errorMessages';
+import { updateTotalReactionsOfAProject } from '../services/reactionsService';
+import { getConnection } from 'typeorm';
+import { logger } from '../utils/logger';
+
+@ObjectType()
+class ToggleResponse {
+  @Field(type => Boolean)
+  reaction: boolean;
+
+  @Field(type => Number)
+  reactionCount: number;
+}
+
+@Resolver(of => Reaction)
+export class ReactionResolver {
+  @Query(returns => [Reaction])
+  async getProjectReactions(
+    @Arg('projectId') projectId: number,
+    @Ctx() { user }: Context,
+  ): Promise<Reaction[]> {
+    return await Reaction.find({
+      where: { projectId: projectId || -1 },
+    });
+  }
+
+  @Mutation(returns => Reaction)
+  async likeProjectUpdate(
+    @Arg('projectUpdateId', type => Int) projectUpdateId: number,
+    @Ctx() { req: { user } }: MyContext,
+  ): Promise<Reaction> {
+    if (!user || !user?.userId)
+      throw new Error(errorMessages.AUTHENTICATION_REQUIRED);
+
+    const queryRunner = getConnection().createQueryRunner();
+
+    await queryRunner.connect();
+
+    await queryRunner.startTransaction();
+
+    const projectUpdate = await queryRunner.manager.findOne(
+      ProjectUpdate,
+      { id: projectUpdateId },
+      { select: ['projectId'] },
+    );
+
+    if (!projectUpdate) throw Error(errorMessages.PROJECT_UPDATE_NOT_FOUND);
+
+    try {
+      const reaction = await queryRunner.manager.create(Reaction, {
+        userId: user?.userId,
+        projectUpdateId,
+        reaction: 'heart',
```

Comment:
Should we move the 'heart' to a constant? I know right now we only have hearts, but maybe will add more in the future. So we don't have heart string hardcoded.
And reuse it for likeProjectUpdate and likeProject.